src/tnlp.cc

```diff
@@ -45,7 +45,7 @@ namespace roboptim
       n = static_cast<Index> (solver_.problem ().function ().inputSize ());
       m = static_cast<Index> (constraintsOutputSize ());
 
-      function_t::vector_t x (n);
+      GenericFunction<EigenMatrixSparse>::vector_t x (n);
```

Comment:
All the typedefs are not always available as a GenericFunction<traits_t>, so it is why I removed the function_t typedef.

I guess writing something like `typedef typename GenericDifferentiableFunction<traits_t> function_t;` would be ok